### ISSUE TITLE
Fill in array with same values

### ISSUE THREAD

**Issue by endragor** [Araq/Nim#2462]
Sometimes you need to fill in the whole array with values different from type defaults. For example, you may have a `bool` array, and you'd like make it have all `true` values, instead of all `false`.
Seems like there is no built-in procedure to do that.

As a workaround, it's possible to use `mapIt` from `sequtils`, but it doesn't look straightforward.

**Comment by def-** [comment]
I don't know, it's not that bad:

```nimrod
var xs: array[20, bool]
for x in xs.mitems: x = true
```

**Comment by endragor** [comment]
```
test.nim(2, 11) Error: undeclared identifier: 'mitems'
```

**Comment by def-** [comment]
Only works in the devel branch, not 0.10.2.

**Comment by endragor** [comment]
OK, thanks. Still, this would look better:
```
xs.fill(true)
```

**Comment by reactormonk** [comment]
I would prefer

    var xs: array[20, bool] = true

**Comment by dom96** [comment]
Perhaps ``sequtils`` could be extended to include array utils as well. But then it should really just be called ``listutils``.

**Comment by endragor** [comment]
`listutils` might get confused with `lists` module that provide linked lists.

**Comment by Araq** [comment]
@reactormonk Your way is too error prone.

**Comment by reactormonk** [comment]
@Araq I agree, but it looks nice.

**Comment by RaphaelHythloday** [comment]
I think this is an essential feature; why can python's `[False] * 20` be used?

**Comment by RaphaelHythloday** [comment]
*can't

**Comment by RaphaelHythloday** [comment]
Also, since when the array is initialized it has to be filled with something anyway, it would be more efficient to initialize it with the necessary value instead of initializing it with the wrong value and then replacing. There should therefore be a option when the array is initialized to fill it with an arbitrary element.

**Comment by borisvassilev** [comment]
How about something allong the lines of

    proc fill[T](var openArray[T], first, last: Natural, value: T)

Or am I missing the point completely? I actually got here because I was looking for the "correct" Nim way to fill (part of) an array with the same values.

**Comment by Varriount** [comment]
@borisvassiliev That seems reasonable.

**Comment by borisvassilev** [comment]
@Varriount I am only not certain about the argument order for the 4-argument version: should the value come last? And would that belong to the algorithm module?

**Comment by borisvassilev** [comment]
Yes, please check out the pull request (https://github.com/Araq/Nim/pull/2701) and comment. This should resolve the issue, if I understood it correctly.

**Comment by dom96** [comment]
Fixed via #2701. Thank you @borisvassilev.
